polaris-react/src/components/DataTable/DataTable.scss

```diff
@@ -259,3 +267,55 @@ $breakpoint: 768px;
     background: none;
   }
 }
+
+.StickyTable {
+  position: relative;
+  top: 0;
+  left: 0;
+  right: 0;
+  visibility: hidden;
+  z-index: var(--p-z-1);
+}
+
+.StickyTableHeader {
+  position: absolute;
+  display: flex;
+  flex-direction: column;
+  width: 100%;
+  overflow: hidden;
+
+  &:not(.StickyTableHeader-isSticky) {
+    top: -1000px;
+    left: -1000px;
+  }
+}
+
+.StickyHeaderCell {
+  flex-shrink: 0;
+}
+
+.StickyTableColumnHeader-isScrolling {
+  box-shadow: 1px 1px 0 0 var(--p-border-divider),
```

Comment:
Should we be using something like `var(--p-border-width-1)` and/or rems of some sort so that the shadow scales with the border/border-radius/etc?